Trigger Claude Code from Port (#2703)
port-labs/port-docs: docs/guides/all/trigger-claude-code-from-port.md

---
displayed_sidebar: null

> description: Learn how to set up Claude Code triggers from Port to enable AI-powered coding assistance in your development workflow with usage tracking.

Reviewer: ```suggestion
description: Learn how to trigger Claude Code from Port to enable AI-powered coding assistance in your development workflow while tracking its usage.
```

---

import GithubActionModificationHint from '/docs/guides/templates/github/_github_action_modification_required_hint.mdx'

# Trigger Claude Code from Port

> This guide demonstrates how to set up Claude Code triggers from Port, enabling AI-powered coding assistance in your development workflow. By leveraging Claude Code, you can significantly reduce manual coding tasks and enhance productivity, allowing developers to focus on more complex problem-solving. You will learn how to create self-service actions that can trigger Claude Code and configure the necessary GitHub workflows to handle the execution process with comprehensive usage tracking.

Reviewer: ```suggestion
This guide demonstrates how to trigger Claude Code from Port, enabling AI-powered coding assistance in your development workflow. By leveraging Claude Code, you can significantly reduce manual coding tasks and enhance productivity, allowing developers to focus on more complex problem-solving. 
You will learn how to create self-service actions that can trigger Claude Code and configure the necessary GitHub workflows to handle the execution process with comprehensive usage tracking.
```

<img src="/img/guides/trigger-claude-code-from-port-flow.jpg" border="1px" width="100%" />

> ## Common use cases

Reviewer: Similar to the trigger copilot, the use cases should be more value focused, for example empower developers with access to claude code from a central place, track the usag of claude code arcoss your organization, triggert claude code as part of workflows such as in response to new bugs

- **Trigger Claude Code for automated code generation** and assistance
- **Enhance code quality** – use Claude Code for refactoring, updates, and documentation, reducing technical debt.

## Prerequisites

This guide assumes the following:
- You have a Port account and have completed the [onboarding process](https://docs.port.io/getting-started/overview).
- [Port's GitHub app](https://docs.port.io/build-your-software-catalog/sync-data-to-catalog/git/github/) is installed in your account.
- You have access to [create and configure AI agents](https://docs.port.io/ai-agents/overview#getting-started-with-ai-agents) in Port.

> - Claude Code is enabled in your repository.

Reviewer: the way I understand it, this guide explains how to enable Claude Code in the repository

- `ANTHROPIC_API_KEY`: Your Anthropic API key for Claude Code access.

### Create Claude Code action

1. Go to the [self-service](https://app.getport.io/self-serve) page of your portal.
2. Click on `+ New Action`.
3. Click on the `{...} Edit JSON` button.
4. Copy and paste the following JSON configuration:

    <details>
    <summary><b>Claude Code action (Click to expand)</b></summary>

    <GithubActionModificationHint/>

    ```json showLineNumbers
    {
      "identifier": "run_claude_code",
      "title": "Run Claude Code",
      "icon": "Code",
      "description": "Open a Claude Code PR on any given repository",
      "trigger": {
        "type": "self-service",
        "operation": "CREATE",
        "userInputs": {
          "properties": {
            "prompt": {
              "type": "string",
              "title": "Prompt",
              "description": "The prompt to pass to Claude Code (AI Coding Agent)",
              "format": "multi-line"
            },
            "service": {

>               "type": "string",

Reviewer: why isn't it an entity selection of repositories?

              "description": "The service associated with the Claude's Code implementation",
              "blueprint": "service",
              "title": "Service",
              "format": "entity"
            }
          },
          "required": [
            "prompt",
            "service"
          ],
          "order": [
            "prompt",
            "service"
          ]
        }
      },
      "invocationMethod": {
        "type": "GITHUB",
        "org": "<GITHUB_ORG>",
        "repo": "<GITHUB_REPO>",
        "workflow": "claude-backend.yaml",
        "workflowInputs": {
          "command": "{{ .inputs.prompt }}",
          "repo_name": "{{ .inputs.service.title }}",
          "run_id": "{{ .run.id }}"
        },
        "reportWorkflowStatus": false
      },
      "requiredApproval": false
    }
    ```
    </details>

5. Click `Save` to create the action.

## Set up GitHub workflow

> Create the file `.github/workflows/claude-backend.yaml` in the `.github/workflows` folder of your repository.

Reviewer: I think we have a general note we say in othre guides around creating a port repository with all the related workflows or somethign like that.

The stup is 1 repo with the backend workflow, and claude code is able to be triggered from all repositores
